Modules/mathmodule.c

```diff
@@ -961,24 +961,16 @@ math_1(PyObject *arg, double (*func) (double), int can_overflow)
         return NULL;
     errno = 0;
     r = (*func)(x);
-    if (Py_IS_NAN(r) && !Py_IS_NAN(x)) {
-        PyErr_SetString(PyExc_ValueError,
-                        "math domain error"); /* invalid arg */
-        return NULL;
-    }
+    if (Py_IS_NAN(r) && !Py_IS_NAN(x))
+        errno = EDOM;
     if (Py_IS_INFINITY(r) && Py_IS_FINITE(x)) {
         if (can_overflow)
-            PyErr_SetString(PyExc_OverflowError,
-                            "math range error"); /* overflow */
+            errno = ERANGE;
         else
-            PyErr_SetString(PyExc_ValueError,
-                            "math domain error"); /* singularity */
-        return NULL;
+            errno = EDOM;
     }
-    if (Py_IS_FINITE(r) && errno && is_error(r))
-        /* this branch unnecessary on most platforms */
+    if (errno && is_error(r))
```

Comment:
I don't understand this well enough to review it — do you know what situation / platform the left hand side code is worried about?

Reply:
Well, this is for some broken libm.  de4296281a shows no glue on which one (if it does exists).  I'll revert this part, however.

Reply:
Oh, I got it.  After 9c32ae0, ``r`` could be infinite (or nan) at this point.  But comment should be restored, of course.

BTW, in some sense 9c32ae0 could be viewed as an unrelated code refactoring (should be helpful for #101410, for example), yet that improved coverage on one line.  Do you suggest to revert this commit?

Reply:
Yeah, if you revert I can merge, otherwise will need to wait for someone else to confirm correctness

Reply:
@mdickinson, maybe you could review that part (second commit, https://github.com/python/cpython/commit/9c32ae022f07ea90892175a2d7f5c777621be0e0)?

Reply:
Ok, @hauntsaninja, https://github.com/python/cpython/commit/9c32ae022f07ea90892175a2d7f5c777621be0e0 was reverted.  Perhaps, this can go to other PR...

The PR description was updated (maybe you can use this for a commit message).